source/override-properties.d.ts

```diff
@@ -25,5 +25,9 @@ type Fizz = OverrideProperties<Foo, {b: number; c: number}>
 */
 export type OverrideProperties<
 	TOriginal,
-	TOverride extends {[Key in keyof TOverride]: Key extends keyof TOriginal ? TOverride[Key] : never},
+	TOverride extends Partial<Record<keyof TOriginal, unknown>> & {
```

Comment:
Can you add a short code comment about why we do this (to get autocomplete).

Reply:
Absolutely! Good call!